als-suggestions/shared/src/main/scala/org/mulesoft/als/suggestions/plugins/aml/webapi/SecuredByCompletionPlugin.scala

```diff
@@ -18,39 +18,66 @@ object SecuredByCompletionPlugin extends AMLCompletionPlugin {
+        getSecurityNames(request)
+      else Nil
     }
   }
 
+  private def isSecurityScalarValue(request: AmlCompletionRequest): Boolean =
+    !isAKey(request) && request.astPartBranch.parentEntryIs("security")
+
+  private def isAKey(request: AmlCompletionRequest): Boolean =
+    request.astPartBranch.isKeyLike
+
   private def isWritingSecuredBy(request: AmlCompletionRequest): Boolean =
-    request.amfObject match {
-      case _: ParametrizedSecurityScheme =>
-        compatibleParametrizedSecurityScheme(request)
-      case _: SecurityRequirement =>
-        (request.fieldEntry.map(_.field).contains(SecurityRequirementModel.Name) || request.fieldEntry.isEmpty) &&
-        underSecurityKey(request) &&
-        (request.astPartBranch.isInArray || request.astPartBranch.isValue)
-      case _: Server => request.fieldEntry.map(_.field).contains(ServerModel.Security)
-      case _         => false
-    }
+    !isSecurityScalarValue(request) &&
+      (request.amfObject match {
+        case _: ParametrizedSecurityScheme =>
+          compatibleParametrizedSecurityScheme(request)
+        case _: SecurityRequirement =>
+          (request.fieldEntry.map(_.field).contains(SecurityRequirementModel.Name) || request.fieldEntry.isEmpty) &&
+          underSecurityKey(request) &&
+          (request.astPartBranch.isInArray || request.astPartBranch.isValue)
+        case _: Server =>
+          request.fieldEntry.map(_.field).contains(ServerModel.Security)
+        case _         => false
+      })
 
   private def compatibleParametrizedSecurityScheme(request: AmlCompletionRequest) =
     request.astPartBranch.isInArray && underSecurityKey(request) && request.fieldEntry.isEmpty && request.amfObject
       .isInstanceOf[ParametrizedSecurityScheme]
 
-  private def getSecurityNames(prefix: String, dp: DeclarationProvider): Seq[RawSuggestion] =
-    new AMLRamlStyleDeclarationsReferences(Seq(SecuritySchemeModel.`type`.head.iri()), prefix, dp, None)
-      .resolve()
+  private def getSecurityNames(request: AmlCompletionRequest): Seq[RawSuggestion] =
+    new AMLRamlStyleDeclarationsReferences(
+          Seq(SecuritySchemeModel.`type`.head.iri()), request.prefix, request.declarationProvider, None
+    ).resolve(rawSuggestionBuilder(request))
 
   private def underSecurityKey(request: AmlCompletionRequest) =
     request.astPartBranch.parentEntryIs("security") ||
       request.astPartBranch.parentEntryIs("securedBy") // use metadata (dialect) here
+
+  private def rawSuggestionBuilder(request: AmlCompletionRequest)(name: String, de: DomainElement): RawSuggestion = {
+    if (isAKey(request) || compatibleParametrizedSecurityScheme(request)) {
+      if (hasScopes(de))
+        RawSuggestion.apply(name, SuggestionStructure(isKey = true, rangeKind = ArrayRange))
+      else
+        RawSuggestion.apply(name, SuggestionStructure(isKey = true, rangeKind = BoolScalarRange)) // En este caso debería autocompletar con corchetes '[]'
```

Comment:
if we're gonna leave commets, let's try to leave it in english

Reply:
Comment removed